packages/turbo/ios/RNSession.swift

```diff
@@ -48,7 +48,7 @@ class RNSession: NSObject {
         return
       }
       newView.becameTopMostView()
-      newView.performRestorationVisit()
+      newView.viewWillAppear()
```

Comment:
I'm sorry for keep digging in this, but we could still improve this. Why is `registerVisitableView` responsible for calling `becameTopMostView`? I think there's too much indirection here.

What about:

```swift
// RNVisitableView
func didBecomeTopMostView(restored: Bool) {
  session.delegate = self
  
  if (restored) {
    turboSession.visitableViewWillAppear(controller)
  }
}

func visitableWillAppear()
  session.registerVisitableView(...)
}

func visitableViewDidAppear() {
  didBecomeTopMostView(restored: false)
}

func visitableWillAppear() {
  session.unregisterVisitableView(...)
}

// RNSession
func registerVisitableView(..) {
  // Only concerned with registering views, not calling back to view, so we'll remove the `didBecomeTopMostView()` - this is instead called from the `visitableViewDidAppear`
}

func unregisterVisitableView() {
  // .. unregister here..

  if (wasTopMostView) {
    newView.didBecomeTopMostView(restored: true)
  }
}
```

This removes the uneeded circular indirection and side-effect in `registerVisitableView` - now that method is only concerned with registering the view. Setting up the delegate etc. is handled in `RNVisitableView` itself.